unidist/core/backends/mpi/core/communication.py

```diff
@@ -129,6 +130,26 @@ def __init__(self, comm):
             self.topology[host][host_rank] = global_rank
             self.host_by_rank[global_rank] = host
 
+        mpi_hosts = MpiHosts.get()
+        if mpi_hosts is not None:
+            host_list = mpi_hosts.split(",") if mpi_hosts is not None else ["localhost"]
+            host_count = len(host_list)
+
+            # check running hosts
+            if self.is_root_process() and len(self.topology.keys()) > host_count:
+                print(
```

Comment:
Should this be an exception?

Reply:
I thought about it, but it can be correct if you run script without "mpiexec" using MPI_HOSTS.
In this case shared memory can't be used, but we have another validation for it.